From 2c87c998801c6ad3259a468ed815f72b2a11e508 Mon Sep 17 00:00:00 2001
From: Javier Pena <jpena@redhat.com>
Date: Thu, 1 Oct 2020 10:59:41 +0200
Subject: [PATCH] Skip tests failing in Python 3.9

Two monkeypatch-related tests are failing in Python 3.9 [1]. While
the cause for the error is figured out, let's skip those two tests
if running in Python 3.9.

[1] - https://github.com/testing-cabal/fixtures/issues/44
---
 fixtures/tests/_fixtures/test_monkeypatch.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/fixtures/tests/_fixtures/test_monkeypatch.py b/fixtures/tests/_fixtures/test_monkeypatch.py
index 6f11fab..8e89b7a 100644
--- a/fixtures/tests/_fixtures/test_monkeypatch.py
+++ b/fixtures/tests/_fixtures/test_monkeypatch.py
@@ -14,6 +14,7 @@
 # limitations under that license.
 
 import functools
+import sys
 
 import testtools
 from testtools.matchers import Is
@@ -181,6 +182,7 @@ class TestMonkeyPatch(testtools.TestCase, TestWithFixtures):
         self._check_restored_static_or_class_method(oldmethod, oldmethod_inst,
                 C, 'foo_cls')
 
+    @testtools.skipIf(sys.version_info > (3, 8), "Fails with Python 3.9")
     def test_patch_classmethod_with_classmethod(self):
         oldmethod = C.foo_cls
         oldmethod_inst = C().foo_cls
@@ -212,6 +214,7 @@ class TestMonkeyPatch(testtools.TestCase, TestWithFixtures):
         self._check_restored_static_or_class_method(oldmethod, oldmethod_inst,
                 C, 'foo_cls')
 
+    @testtools.skipIf(sys.version_info > (3, 8), "Fails with Python 3.9")
     def test_patch_classmethod_with_boundmethod(self):
         oldmethod = C.foo_cls
         oldmethod_inst = C().foo_cls
-- 
2.26.2

